src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
 ```mermaid
-flowchart LR
-    nodeA1(Parameter) --> nodeA6(Load)
-    nodeA6(Load) --> nodeA2(Add)
-    nodeA3(Parameter) --> nodeA7(Load)
-    nodeA7(Load) --> nodeA2(Add)
-    nodeA2(Add) --> nodeA8(Store)
-    nodeA8(Store) --> nodeA5(Result)
-classDef carbon1 fill:#E9E9E9, stroke: #AEAEAE, color: #262626
-classDef moss1 fill:#D7F3A2, stroke: #B1D272, color: #262626
+ graph LR
+   Source[nGraph \n model]
+   subgraph Snippets
+      direction LR
+      subgraph Optimizer[Optimizer]
+        direction LR
+        Data[Data flow \n optimizations]
+        Control[Control flow \n optimizations]
+        Data-->|nGraph \nIR|Control
+      end
+      Frontend[Tokenizer]-->|nGraph \nIR|Data
+      Control-->|Linear \nIR|Backend[Generator]
+   end
+   Source --> Frontend
+   Backend --> Executable[Kernel]
+classDef no-bg-color fill:none,stroke-width:0px
 classDef steel1 fill:#B9D6E5, stroke: #86B3CA, color: #262626
 classDef daisy1 fill:#FFE17A, stroke: #FEC91B, color: #262626
-class nodeA2 daisy1
-class nodeA5 moss1
-class nodeA8 carbon1
-class nodeA1,nodeA3,nodeA6,nodeA7 steel1
+class Frontend,Optimizer,Backend steel1
+class Source,Executable daisy1
 ```
+Instead of a source code, `Snippets` take `nGraph` model as an input. Then the `Tokenizer` (which is essentially a `Snippets` `Frontend`) parses an input `nGraph model`, and tries to find a part of the model that could be processed by `Snippets`. If such a part is found, `Tokenizer` converts it to an `nGraph IR` and stores inside a `Subgraph` node. `nGraph IR` - is one of the two `IR` types used by `Snippets`, it is simply a small `nGraph model` that can contain `Snippets`-specific operations. `nGraph IR` is then passed to the `Optimizer` unit that in turn consists of two subunits. The purpose of the first subunit is to perform data flow optimizations, whereas the second subunit is focused on control flow optimizations. Note that the control flow optimizations are performed on `Linear IR` which was especially designed for this purpose. Finally, the optimized `Linear IR` is used by the `Generator` (which is `Snippets` `Backend`) to produce executable code, which we will refer to as `Kernel`. As discussed in the Introduction, the purpose of the `Kernel` is to process a part of the initial tensor, and several `Kernels` are usually executed in parallel to process the whole tensor. Note that a `Kernel` usually applies several operations (`Op instructions` blocks from the Introduction), the exact number depends on the source model topology: sometimes it's just a couple, but could more than twenty `Ops`.
```

Comment:
I'm not sure but maybe there can be questions how it's stored
`If such a part is found, `Tokenizer` converts it to an nGraph IR and stores inside a Subgraph node.` -> `If such a part is found, `Tokenizer` converts it to an nGraph IR and stores inside a Subgraph node as small nGraph model which is called body.`

Reply:
I've intentionally tried to minimize the amount of details here, since this section is dedicated to an overall architecture, and too many details could be overwhelming. In fact, I'm not even sure that it's worth to mention two IR types and data/control flow optimization blocks at this point.
So I think it's better to leave the `body` for the optimizations section